### PR TITLE
M3/M4 samples

### PR DESCRIPTION
**Pull request by stevengum97**
## Add new M3 samples:
- [X] conversation-update-es6
- [X] echobot-console-es6
- [X] single-prompt-bot-es6
- [X] multiple-prompts-bot-es6
- [X] rich-cards-es6
- [X] dispatch-es6
- [X] luis-bot-es6 // Example already exists
- [x] qnamaker-bot-es6
- [x] echobot-es6-botframework-webchat

____
## Cleanup:
 - [X] Remove preexisting samples
 - [X] Left echobot-es6 as part of parity with [dotnet](https://github.com/Microsoft/botbuilder-dotnet/tree/master/samples-final/AspNetCore-EchoBot-With-State)
 - [X] Left [dialogs](https://github.com/Microsoft/botbuilder-js/tree/9fe24a9240ced2da52d42abf588498af2c7d078d/samples/dialogs) per conversation with Steve

### CODE REVIEW

**Comment by stevengum97** [comment]
@CezaryMarcjan and @DeniseMak, can you take a look at the [dispatch sample][dispatch] and let me know if you have any feedback regarding it? 

Per the original email thread that outlines the samples to be added to the repo, I was planning on going back in and re-adding the QnAMaker KBID and Subscription Key for quick spin-up as QnAMaker does not have a "[public][luis-public]" setting at this time.

  [dispatch]: https://github.com/Microsoft/botbuilder-js/pull/208/commits/aa37e6a7a014efd4df12ffbfa01e908fc8187be3
  [luis-public]: https://westus.dev.cognitive.microsoft.com/docs/services/5890b47c39e2bb17b84a55ff/operations/58aeface39e2bb03dcd5909e

File: samples/conversation-update-es6/app.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+});
+
+// Listen for incoming requests 
+server.post('/api/messages/', (req, res) => {
+    adapter.processActivity(req, res, (context) => {
+        // On "conversationUpdate"-type activities this bot will send a greeting message to users joining the conversation.
+        if (context.activity.type === 'conversationUpdate' && context.activity.membersAdded[0].name !== 'Bot') {
+            return context.sendActivity(`Hello "${context.activity.membersAdded[0].name}"!`);
+        }
+        if (context.activity.type === 'message') {
```
Comment (Stevenic) [inline samples/conversation-update-es6/app.js]: >if [](start = 8, length = 2)

why not an "else if" ?

File: samples/conversation-update-es6/app.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    appId: process.env.MICROSOFT_APP_ID, 
+    appPassword: process.env.MICROSOFT_APP_PASSWORD 
+});
+
+// Listen for incoming requests 
+server.post('/api/messages/', (req, res) => {
+    adapter.processActivity(req, res, (context) => {
+        // On "conversationUpdate"-type activities this bot will send a greeting message to users joining the conversation.
+        if (context.activity.type === 'conversationUpdate' && context.activity.membersAdded[0].name !== 'Bot') {
+            return context.sendActivity(`Hello "${context.activity.membersAdded[0].name}"!`);
```
Comment (Stevenic) [inline samples/conversation-update-es6/app.js]: >return [](start = 12, length = 6)

I think we've decided to use async/await in all the samples

File: samples/dispatch-es6/app.js [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+// To create a new service, upload this knowledge base by going to QnA Maker at https://qnamaker.ai/Create
+const faq = new QnAMaker({
+    knowledgeBaseId: '',
+    subscriptionKey: '',
+    serviceEndpoint: 'https://westus.api.cognitive.microsoft.com/'
+});
+
+// create conversation state
+const conversationState = new ConversationState(new MemoryStorage());
+adapter.use(conversationState);
```
Comment (Stevenic) [inline samples/dispatch-es6/app.js]: Cash and I have decided that we should always show the use of both UserState() and ConversationState() in all samples with state. I can send you the general boilerplate to include

File: samples/dispatch-es6/app.js [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+                    const weatherResults = await weather.recognize(context);
+                    const topWeatherIntent = LuisRecognizer.topIntent(weatherResults);
+                    await dc.begin(topWeatherIntent, weatherResults);
+                    break;
+                case 'q_faq':
+                    await faq.answer(context);
+                    break;
+                default:
+                    await dc.begin('None');
+            }
```
Comment (Stevenic) [inline samples/dispatch-es6/app.js]: This should be restructured a bit. I'll talk to you about this

File: samples/conversation-update-es6/app.js (hunk 1 above)
Comment (stevengum97) [reply samples/conversation-update-es6/app.js]: I think I learned that from someone at CSE... But I will change it 👍

File: samples/conversation-update-es6/app.js (hunk 2 above)
Comment (stevengum97) [reply samples/conversation-update-es6/app.js]: I recall that now, I'll go in and change this.

File: samples/dispatch-es6/app.js (hunk 4 above)
Comment (stevengum97) [reply samples/dispatch-es6/app.js]: I believe I covered the changes we discussed, which was centered on how to call `dc.continue()`; I based the new changes on the sample code in [`samples/dialogs/alarmbot-ts`](https://github.com/Microsoft/botbuilder-js/tree/9fe24a9240ced2da52d42abf588498af2c7d078d/samples/dialogs/alarmbot-ts).

**Pull request merged**